lib/api/core/models/repositories/repository.js

```diff
@@ -186,69 +181,68 @@ class Repository {
    *
    * The opts object currently accepts one optional parameter: key, which forces
    * the cache key to fetch.
-   * In case the key is not provided, it defauls to <collection>/id, i.e.: _kuzzle/users/12
+   * In case the key is not provided, it defauls to <collection>/id, i.e.: this.kuzzle/users/12
    *
-   * @param {string} id The id of the object to get
-   * @param {object} [opts] Optional options.
+   * @param {string} id - The id of the object to get
+   * @param {object} [options] - Optional options.
    * @returns {Promise}
    */
-  load (id, opts) {
+  load (id, options = {}) {
     if (!this.cacheEngine) {
-      return this.loadOneFromDatabase(id);
+      return this.loadOneFromDatabase(id, this.collection);
     }
 
-    return this.loadFromCache(id, opts)
+    return this.loadFromCache(id, options, this.collection)
       .then(object => {
         if (object === null) {
           if (this.databaseEngine === null) {
             return null;
           }
 
-          return this.loadOneFromDatabase(id)
+          return this.loadOneFromDatabase(id, this.collection)
             .then(objectFromDatabase => {
               if (objectFromDatabase !== null) {
-                this.persistToCache(objectFromDatabase);
+                this.persistToCache(objectFromDatabase, this.collection);
```

Comment:
`collection` parameter is not used in `persistToCache` method
(same remark for all other calls of this within the file)